### PR TITLE
boards: added support for ST nucleo-f091

### PR DESCRIPTION
**Pull request by haukepetersen**
~~rebased on #2593~~

Needed a board for testing the xbee driver and this was only one I had laying around... So here is the port :-)

### CODE REVIEW

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
Added support for GPIO and rebased on #2595.

File: boards/nucleo-f091/Makefile.features [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FEATURES_PROVIDED += cpp
+FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_gpio
```
Comment (LudwigOrtmann) [inline boards/nucleo-f091/Makefile.features]: rebase on master and add `periph_cpuid`

Comment (haukepetersen) [inline boards/nucleo-f091/Makefile.features]: done.

**Comment by haukepetersen** [comment]
added feature `PERIPH_CPUID` and rebased to master.

File: boards/nucleo-f091/dist/openocd.cfg [hunk 2]
```diff
@@ -0,0 +1 @@
+source [find board/st_nucleo_f030r8.cfg]
```
Comment (thomaseichinger) [inline boards/nucleo-f091/dist/openocd.cfg]: Please check for `board/st_nucleo_f0.cfg`.

File: boards/nucleo-f091/include/periph_conf.h [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+#define UART_0_TX_PIN       2
+#define UART_0_AF           1
+
+/* UART 1 device configuration */
+#define UART_1_DEV          USART1
+#define UART_1_CLKEN()      (RCC->APB2ENR |= RCC_APB2ENR_USART1EN)
+#define UART_1_CLKDIS()     (RCC->APB2ENR &= (~RCC_APB2ENR_USART1EN))
+#define UART_1_IRQ          USART1_IRQn
+#define UART_1_ISR          isr_usart1
+/* UART 0 pin configuration */
```
Comment (thomaseichinger) [inline boards/nucleo-f091/include/periph_conf.h]: s/0/1/

**Comment by haukepetersen** [comment]
rebased and addressed comments.

**Comment by thomaseichinger** [comment]
ACK when Travis passed.

**Comment by haukepetersen** [comment]
fixed dos line endings in the STM header file.

**Comment by haukepetersen** [comment]
Travis is happy, so go

**Pull request merged**